src/Layout.ts

```diff
@@ -1736,23 +1635,21 @@ export class Union extends Layout {
    * value is an instance of that variant, with no explicit
    * discriminator.  Otherwise the {@link Union#defaultLayout|default
    * layout} is used to decode the content. */
-  decode(b: Uint8Array, offset?: number) {
-    if (undefined === offset) {
-      offset = 0;
-    }
+  decode(b: Uint8Array, offset = 0): LayoutObject {
     let dest: LayoutObject;
     const dlo = this.discriminator;
     const discr = dlo.decode(b, offset);
     const clo = this.registry[discr];
     if (undefined === clo) {
-      const defaultLayout = this.defaultLayout as any;
+      const defaultLayout = this.defaultLayout;
       let contentOffset = 0;
       if (this.usesPrefixDiscriminator) {
         contentOffset = (dlo as UnionLayoutDiscriminator).layout.span;
       }
       dest = this.makeDestinationObject();
       dest[dlo.property] = discr;
-      dest[defaultLayout.property] = defaultLayout.decode(b, offset + contentOffset);
+      // @FIXME: defaultLayout and defaultLayout.property aren't guaranteed
+      dest[defaultLayout!.property!] = defaultLayout!.decode(b, offset + contentOffset);
```

Comment:
Removed cast as `any` above, but we're still assuming that `defaultLayout` is non-null and that it has a `property`